front/components/assistant/conversation/AgentMessage.tsx

```diff
@@ -202,9 +202,12 @@ export function AgentMessage({
               chainOfThoughts: [event.chainOfThought],
             };
           }
-          // Otherwise, we don't do anything, because we already have this chain of thought in the chainOfThoughts array
-          // (and it wasn't being streamed as regular tokens, so we don't need to clear the content)
-          return m;
+          // Otherwise, we insert an empty COT so that any future COT (if any) gets appended to that one.
```

Comment:
would explain there a tad more because it took me some time to understand
would change "so that any future..." to `so that any future COT **tokens, produced by "generation_tokens" event below,** gets appended to that one"

would add a comment on top of the 'agent_cot_' switch case
"Event generated when a new chain of thought generation is started---except when the CoT was streamed as regular tokens in which case we only detect it at the end"